server/webapp/WEB-INF/rails/webpack/models/users/users.ts

```diff
@@ -108,6 +118,45 @@ export class User {
   pluginRoles() {
     return _(this.roles()).filter((role) => "gocd" !== role.type).map((role) => role.name).value();
   }
+
+  markUpdateInprogress() {
+    this.updateOperationStatus(UpdateOperationStatus.IN_PROGRESS);
+  }
+
+  markUpdateSuccessful() {
+    this.updateOperationStatus(UpdateOperationStatus.SUCCESS);
+  }
+
+  markUpdateUnsuccessful() {
+    this.updateOperationStatus(UpdateOperationStatus.ERROR);
+  }
+
+  clearUpdateStatus() {
+    this.updateOperationStatus(null);
+  }
+
+  updateFromJSON(json: UserJSON) {
```

Comment:
Lets not mutate user objects, replacing is a better option. If a new attribute is added to a user, then someone needs to remember to update this method as well.